File: server/battle/src/main/java/info/chenliang/moba/message/BattleToPlayerStub.java
package info.chenliang.moba.message;
import java.io.*;
import info.chenliang.talky.*;
public abstract class BattleToPlayerStub {
protected abstract DataOutputStream begin();
protected abstract void end();
public void battlePing() throws Exception{
DataOutputStream dos = begin();
dos.write(0);
end();
}
public void battlePong() throws Exception{
DataOutputStream dos = begin();
dos.write(1);
end();
}
public void onJoinArena(int arenaId,Sync sync,int entityId) throws Exception{
DataOutputStream dos = begin();
dos.write(2);
dos.writeInt(arenaId);
sync.serialize(dos);
dos.writeInt(entityId);
end();
}
public void sync(Sync sync) throws Exception{
DataOutputStream dos = begin();
dos.write(3);
sync.serialize(dos);
end();
}
public void playerSpawned(int entityId,SyncItem item) throws Exception{
DataOutputStream dos = begin();
dos.write(4);
dos.writeInt(entityId);
item.serialize(dos);
end();
}
}

File: client/Assets/Scripts/Message/info/chenliang/moba/message/BattleToPlayerStub.cs
namespace info.chenliang.moba.message{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public abstract class BattleToPlayerStub {
protected abstract MemoryEndianBinaryWriter Begin();
protected abstract void End();
public void battlePing(){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(0);
End();
}
public void battlePong(){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(1);
End();
}
public void onJoinArena(Int32 arenaId,Sync sync,Int32 entityId){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(2);
writer.WriteInt32(arenaId);
sync.Serialize(writer);
writer.WriteInt32(entityId);
End();
}
public void sync(Sync sync){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(3);
sync.Serialize(writer);
End();
}
public void playerSpawned(Int32 entityId,SyncItem item){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(4);
writer.WriteInt32(entityId);
item.Serialize(writer);
End();
}
}
}
